contrib_bots/lib/commute_bot.py

```diff
@@ -0,0 +1,228 @@
+        # structure for more aethestic coding
+        for item in content_list:
+            # enables key value pair
+            org = item.split('=')
+            # ensures that invalid inputs are not entered into url request
+            if len(org) != 2:
+                continue
+            key, value = org
+            params.update({key: value})
+
+        # limited to UTC timezone because of difficulty with user inputting
+        # correct string for input
+        if 'departure_time' in params:
+            if params['departure_time'] != 'now':
+                departure_time = params['departure_time']
+                times = departure_time.split(',')
+                times = [int(x) for x in times]
+                # UNIX time from January 1, 1970 00:00:00
+                unix_start_date = dt.datetime(1970, 1, 1, 0, 0, 0)
+                requested_time = dt.datetime(*times)
+                total_seconds = str(int((requested_time-unix_start_date)
+                                        .total_seconds()))
+                params.update({'departure_time': total_seconds})
+
+        elif 'arrival_time' in params:
+            arrival = params['arrival_time']
+            times = arrival.split(',')
+            times = [int(x) for x in times]
+            # UNIX time from January 1, 1970 00:00:00
+            unix_start_date = dt.datetime(1970, 1, 1, 0, 0, 0)
+            requested_time = dt.datetime(*times)
+            total_seconds = str(int((requested_time-unix_start_date)
+                                    .total_seconds()))
+            params.update({'arrival_time': total_seconds})
+
+        self.get_api_key()
+
+        # sends url api request and converts response into json format
+        r = requests.get('https://maps.googleapis.com/maps/api/'
+                         + 'distancematrix/json',
+                         params=params)
+        result = self.validate_requests(r, message, client)
+        if not result:
+            return
+
+        rjson = result
+
+        # JSON list of output variables
+        variable_list = rjson["rows"][0]["elements"][0]
+
+        # determines if user has valid inputs
+        try:
+            not_found = (variable_list["status"] == "NOT_FOUND")
+            invalid_request = (rjson["status"] == "INVALID_REQUEST")
+            no_result = (variable_list["status"] == "ZERO_RESULTS")
+
+            if no_result:
+                self.send_info(message,
+                               "Zero results\nIf stuck, try '@commute help'.",
+                               client)
+                return
+
+            elif not_found or invalid_request:
+                raise IndexError
+        except IndexError:
+            self.send_info(message,
+                           "Invalid input, please see instructions."
+                           "\nIf stuck, try '@commute help'.", client)
+            return
+
+        # origin and destination strings
```

Comment:
I think you should add `mode` here, since the request works without providing a mode.